packages/grafana-llm-app/src/components/AppConfig/AppConfig.tsx

```diff
@@ -14,6 +14,23 @@ import { OpenAISettings } from './OpenAI';
 import { VectorConfig, VectorSettings } from './Vector';
 ///////////////////////
 
+const validateUrl = (url: string): [boolean, string] => {
```

Comment:
`URL.canParse` might be better than something hand-rolled?

Reply:
Makes sense, still need some custom rules to handle diffs between go validation and js validation
e.g. JS
```js
URL.canParse("http://api.open ai.com")
> true
```
Golang
```go
u, err := url.Parse("http://api.open ai.com")
> panic: parse "http://api.open ai.com": invalid character " " in host name
```

Reply:
```javascript
URL.canParse("http://api.open ai.com")
> true
```

geez...